packages/loader/container-loader/src/connectionStateHandler.ts

```diff
@@ -230,9 +235,16 @@ class ConnectionStateCatchup extends ConnectionStateHandlerPassThrough {
 				this.catchUpMonitor?.dispose();
 				this.catchUpMonitor = undefined;
 				break;
+			case ConnectionState.EstablishingConnection:
+				assert(
+					this._connectionState === ConnectionState.Disconnected ||
+						this._connectionState === ConnectionState.EstablishingConnection,
+					"connectivity transition to establishing connection",
```

Comment:
Curious, why we could be already in ConnectionState.EstablishingConnection mode? Is it a reflection that it already changed (then I guess we should not see ConnectionState.Disconnected cases), or we actually could come here in those two modes? Might be worth adding some comment on the state matrix / transitions as it's hard to figure it out on the spot